lib/Service.js

```diff
@@ -131,8 +131,10 @@ class Service extends Component {
         delete this._pendingRequests[id];
 
         // ** Check for errors
-        if (res.error)
-            return req.fail(res.error);
+        if (res.error){
+            var err = errors.isError(res.error) ? res.error.toObject() : res.error;
```

Comment:
or perhaps even the following:
` var error = util.isFunction(res.error.toObject) ? res.error.toObject() : res.error; `